tests/systemd/services_tests.py

```diff
@@ -0,0 +1,182 @@
+def account(alias: str):
+    # Generate baker key
+    proc_call(f"sudo -u tezos octez-client gen keys {alias} --force")
+    try:
+        yield alias
+    finally:
+        proc_call(f"sudo -u tezos octez-client forget address {alias} --force")
+
+
+def retry(action, name: str, retry_count: int = 20) -> bool:
+    if action(name):
+        return True
+    elif retry_count == 0:
+        return False
+    else:
+        sleep(5)
+        return retry(action, name, retry_count - 1)
+
+
+def check_running_process(process_name: str) -> bool:
+    return retry(lambda x: x in [proc.name() for proc in process_iter()], process_name)
+
+
+def check_active_service(service_name: str) -> bool:
+    return retry(
+        lambda x: Unit(x.encode(), _autoload=True).Unit.ActiveState == b"active",
+        service_name,
+    )
+
+
+def generate_identity(network):
+    if not os.path.exists(f"/var/lib/tezos/{network}/identity.json"):
+        proc_call(
+            f"sudo -u tezos octez-node identity generate 1 --data-dir /var/lib/tezos/{network}"
+        )
+
+
+def node_service_test(network: str, rpc_endpoint="http://127.0.0.1:8732"):
+    generate_identity(network)
+    with unit(f"tezos-node-{network}.service") as _:
+        # checking that service started 'tezos-node' process
+        assert check_running_process("octez-node")
+        # checking that node is able to respond on RPC requests
+        assert retry(url_is_reachable, f"{rpc_endpoint}/chains/main/blocks/head")
+
+
+def baking_service_test(network: str, protocols: List[str], baker_alias="baker"):
+    with account(baker_alias) as _:
+        generate_identity(network)
+        with unit(f"tezos-baking-{network}.service") as _:
+            assert check_active_service(f"tezos-node-{network}.service")
+            assert check_running_process("octez-node")
+            for protocol in protocols:
+                assert check_active_service(
+                    f"tezos-baker-{protocol.lower()}@{network}.service"
+                )
+                assert check_running_process(f"octez-baker-{protocol}")
+
+
+signer_unix_socket = '"/tmp/signer-socket"'
+
+signer_backends = {
+    "http": "http://localhost:8080/",
+    "tcp": "tcp://localhost:8000/",
+    "unix": f"unix:{signer_unix_socket}?pkh=",
+}
+
+
+def signer_service_test(service_type: str):
+    with unit(f"tezos-signer-{service_type}.service") as _:
+        assert check_running_process(f"octez-signer")
+        proc_call(
+            "sudo -u tezos octez-signer -d /var/lib/tezos/signer gen keys remote --force"
+        )
+        remote_key = get_key_address("-d /var/lib/tezos/signer", "remote")[1]
+        proc_call(
+            f"octez-client import secret key remote-signer {signer_backends[service_type]}{remote_key} --force"
+        )
+        proc_call("octez-client --mode mockup sign bytes 0x1234 for remote-signer")
+
+
+def test_node_mainnet_service():
+    node_service_test("mainnet")
+
+
+def test_node_limanet_service():
```

Comment:
limanet has been decommissioned at this point, can you update these to mumbainet instead?